src/mower_logic/src/mower_logic/behaviors/UndockingBehavior.cpp

```diff
@@ -55,37 +55,45 @@ Behavior *UndockingBehavior::execute() {
   docking_pose_stamped_front.pose = pose.pose.pose;
   docking_pose_stamped_front.header = pose.header;
 
-  int undock_point_count = config.undock_distance * 10.0;
+  int undock_point_count = 10;
+  double incremental_distance = config.undock_distance / undock_point_count;
   for (int i = 0; i < undock_point_count; i++) {
-    docking_pose_stamped_front.pose.position.x -= cos(yaw) * 0.1;
-    docking_pose_stamped_front.pose.position.y -= sin(yaw) * 0.1;
+    docking_pose_stamped_front.pose.position.x -= cos(yaw) * incremental_distance;
+    docking_pose_stamped_front.pose.position.y -= sin(yaw) * incremental_distance;
     path.poses.push_back(docking_pose_stamped_front);
   }
 
   double angle;
-  if (config.undock_fixed_angle)
-    angle = config.undock_angle * (M_PI + M_PI) / 360.0;
-  else
-    angle = next_undock_angle;
+  if (config.undock_fixed_angle) {
+    angle = config.undock_angle * M_PI / 180.0;
+    ROS_INFO_STREAM("Fixed angle undock: " << config.undock_angle);
+  } else {
+    // seed based on first undock time rather than boot so should be ok even without RTC
+    if (seedRqd) {
+      srand(ros::Time::now().toSec());
+      ROS_INFO_STREAM("Random angle undock: Seeded rand()");
+      seedRqd = false;
+    }
+    double ranNum = (((((double)rand()) / RAND_MAX) - 0.5) * 2.0);
+    double ranAngle = abs(config.undock_angle) * ranNum;
+    ROS_INFO_STREAM("Random angle undock: " << ranAngle);
+    angle = ranAngle * (M_PI + M_PI) / 360.0;
+  }
 
-  undock_point_count = config.undock_angled_distance * 10.0;
+  undock_point_count = 10;
+  incremental_distance = config.undock_angled_distance / undock_point_count;
   for (int i = 0; i < undock_point_count; i++) {
-    double orientation;
-    if (config.undock_use_curve)
-      orientation = yaw + ((i + 1) * angle / undock_point_count);
-    else
-      orientation = yaw + angle;
-    docking_pose_stamped_front.pose.position.x -= cos(orientation) * 0.1;
-    docking_pose_stamped_front.pose.position.y -= sin(orientation) * 0.1;
-    tf2::Quaternion q(0.0, 0.0, orientation);
+    double orientation = yaw + angle * (config.undock_use_curve ? ((i + 1) / undock_point_count) : 1);
```

Comment:
I still feel that starting the loop with 1 would be easier, but I don't insist on it. Do overcome the integer division problem, `i + 1.0` should work.

Reply:
Done.